x-pack/packages/kbn-elastic-assistant/impl/assistant/prompt_editor/system_prompt/select_system_prompt/index.tsx

```diff
@@ -174,20 +188,55 @@ const SelectSystemPromptComponent: React.FC<Props> = ({
               onBlur={handleOnBlur}
               options={[...options, addNewSystemPrompt]}
               placeholder={i18n.SELECT_A_SYSTEM_PROMPT}
-              valueOfSelected={selectedPrompt?.id ?? allSystemPrompts[0]?.id}
+              valueOfSelected={valueOfSelected}
+              prepend={isFlyoutMode ? 'Select prompt' : undefined}
             />
           </EuiFormRow>
         )}
       </EuiFlexItem>
 
-      <EuiFlexItem grow={false}>
-        {isEditing && isClearable && (
+      <EuiFlexItem
+        grow={false}
+        css={
+          isFlyoutMode
+            ? css`
+                position: absolute;
+                right: 36px;
+              `
+            : undefined
+        }
+      >
+        {isEditing && isClearable && selectedPrompt && (
           <EuiToolTip content={i18n.CLEAR_SYSTEM_PROMPT}>
             <EuiButtonIcon
               aria-label={i18n.CLEAR_SYSTEM_PROMPT}
               data-test-subj="clearSystemPrompt"
               iconType="cross"
               onClick={clearSystemPrompt}
+              css={
+                isFlyoutMode
+                  ? // mimic EuiComboBox clear button
```

Comment:
Surprised `EuiSuperSelect` doesn't have an `isClearable` attr like `EuiComboBox` -- nice fix!

Text does float behind, but you can hardly tell and it doesn't interfere with clicks. Keyboard nav works like a charm too, so LGTM! 👍 


<img width="541" alt="image" src="https://github.com/elastic/kibana/assets/2946766/a156d664-e468-4911-9d03-2db09fd34cfb">

Reply:
added padding to make sure the button doesn't overflow the text, great catch @spong 🙇